web/html/src/manager/virtualization/guests/guest-properties.js

```diff
@@ -0,0 +1,120 @@
+const React = require("react");
+const ReactDOM = require("react-dom");
+const {SmallPanel} = require("../../../components/panel");
+const {Form, FormGroup, Text, Select} = require("../../../components/input");
+const {SubmitButton, Button} = require("../../../components/buttons");
+const Validation = require("../../../components/validation");
+const {Messages} = require("../../../components/messages");
+const MessagesUtils = require("../../../components/messages").Utils;
+const PropTypes = React.PropTypes;
+
+/*
+ * Component editing a virtual machine properties
+ *
+ * Properties:
+ *   + host
+ *   + submitText
+ *   + submit: a function that takes the hostid, the guest uuid, the form model,
+ *             and a setMessages function as parameters
+ *   + getInitialModel: a function that returns the default values for the model
+ */
+class GuestProperties extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      model: props.getInitialModel(),
+      isInvalid: false,
+      messages: null,
+    };
+
+    this.clearFields = () => {
+      this.setState({
+        model: this.props.getInitialModel()
+      });
+    };
+
+    this.onValidate = (isValid) => {
+      var messages = null;
+      if (!Number.isNaN(Number.parseInt(this.state.model.vcpu)) && (this.state.model.vcpu > this.props.host.cpu.count)) {
+        messages = MessagesUtils.warning("Overcommitting CPU can harm performances.");
+      }
+
+      this.setState({
+        isInvalid: !isValid,
+        messages : messages
+      });
+    };
+
+    this.onChange = (model) => {
+      this.setState({
+        model: model
+      });
+    };
+
+    this.onSubmit = () => {
+      const model = this.state.model;
+      this.props.submit(model, this.setMessages);
+    };
+
+    this.setMessages = (messages) => {
+      this.setState({
+        messages : messages
+      });
+    };
+
+    this.renderMessages = () => {
+      var messages = Array.isArray(this.state.messages) ? this.state.messages : [this.state.messages] || [];
+      if (Array.isArray(this.props.messages)) {
```

Comment:
1st issue: [this.state.messages] || []

The first expression will be always true even if the variable messages is undefined it will be [undefined]

I would simplify all these lines like this:

```
var messages = [].concat(this.state.messages, this.props.messages);
```

Concat can either receive an array/single value or a list of arrays/single values.